docs/auth/auth0/provider.md

```diff
@@ -0,0 +1,74 @@
+---
+id: provider
+title: Auth0 Authentication Provider
+sidebar_label: Auth0
+description: Adding Auth0 as an authentication provider in Backstage
+---
+
+The Backstage `core-api` package comes with an Auth0 authentication provider
+that can authenticate users using OAuth.
+
+## Create an Auth0 Application
+
+1. Log in to the [Auth0 dashboard](https://manage.auth0.com/dashboard/)
+2. Navigate to **Applications**
+3. Create an Application
+   - Name: Backstage (or your custom app name)
+   - Application type: Single Page Web Application
+4. Click on the Settings tab
+5. Add under `Application URIs` > `Allowed Callback URLs`:
+   `http://localhost:7000/api/auth/auth0/handler/frame`
```

Comment:
While technically correct, this infers that the backend is fully visible to the public Internet doesn't it? And also fully visible without any authentication of any kind, since the backend doesn't require auth?  Seems risky, though this is the first I've considered it.

This example seems like only one we might recommend for dev setup, but not for production usage - would that be valid?  I don't have a specific point of change to recommend, but only that a new user may not understand the nuances of what this means is all.

Reply:
I don't think that's quite correct. This OAuth config is whitelisting your callback URL so that the provider allows redirecting there. The actual redirect happens in the browser, so the URL only needs to be accessible for the user signing in. This works fine with VPN setups, etc. where the URL is not publicly internet-accessible. That's also why it works fine with `localhost` and doesn't require exposing your laptop to the internet.

Reply:
Ah ok; gotcha.  Thinking we still may want to documenting some architectural recommendations so each team doesn't try to reinvent the wheel, but what you said makes sense. (essentially, unauth but behind a team's own network)